src/Config.hs

```diff
@@ -0,0 +1,223 @@
+module Config
+  ( Options(..)
+  , MilliSeconds(..)
+  , parseOptionsFromEnvAndCli
+  ) where
+
+import Control.Applicative ((<*>), (<|>))
+import Data.Monoid ((<>))
+
+import Options.Applicative (value, long, auto, option, metavar, help, flag,
+                            str, argument, many, strOption, (<**>))
+
+import qualified Options.Applicative as OptParse
+import qualified Options.Applicative.Builder.Internal as OptParse
+import qualified Text.Read as Read
+
+-- ! Type alias for enviornment variables, used for readability in this module.
+type EnvVar = (String, String)
+
+-- ! Newtype wrapper for millisecond values.
+newtype MilliSeconds = MilliSeconds { unMilliSeconds :: Int }
+  deriving (Eq, Show)
+
+-- | @Options@ contains all the configuration we support in vaultenv. It is
+-- used in our @Main@ module to specify behaviour.
+data Options = Options
+  { oVaultHost       :: String
+  , oVaultPort       :: Int
+  , oVaultToken      :: String
+  , oSecretFile      :: FilePath
+  , oCmd             :: String
+  , oArgs            :: [String]
+  , oNoConnectTls    :: Bool
+  , oNoValidateCerts :: Bool
+  , oNoInheritEnv    :: Bool
+  , oRetryBaseDelay  :: MilliSeconds
+  , oRetryAttempts   :: Int
+  } deriving (Eq, Show)
+
+-- | Behaviour flags that we allow users to set via environment variables.
```

Comment:
s/Behaviour/Behavior (to be consistent with yourself further down)

Reply:
Fixed